### PR TITLE
ConditionalSetValue: Allow use of Bool, UInt8, Int8 as the mask array

### PR DESCRIPTION
**Pull request by imikejackson**
Update unit test to include UInt8 and Int8 Input arrays.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

### CODE REVIEW

**Comment by imikejackson** [comment]
https://github.com/BlueQuartzSoftware/DREAM3DNX/pull/53 is dependent on this PR

@mmarineBlueQuartz

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 1]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
   REQUIRE(RequireDataArrayEqualZero(*ciDataArray));
 }
 
+TEST_CASE("ConditionalSetValue: Test Algorithm UInt8", "[ConditionalSetValue]")
+{
+  DataStructure dataGraph = UnitTest::CreateDataStructure();
+  DataPath ebsdScanPath = DataPath({k_SmallIN100, k_EbsdScanData});
+  DataPath geomPath = DataPath({k_SmallIN100, k_EbsdScanData, k_ImageGeometry});
+  std::shared_ptr<ImageGeom> imageGeometry = dataGraph.getSharedDataAs<ImageGeom>(geomPath);
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:144]:
```suggestion
  const ImageGeom& imageGeometry = dataGraph.getDataRefAs<ImageGeom>(geomPath);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 2]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+  complex::SizeVec3 imageGeomDims = imageGeometry->getDimensions();
+
+  DataPath ciDataPath = DataPath({k_SmallIN100, k_EbsdScanData, k_ConfidenceIndex});
+  DataObject* ciDataObject = dataGraph.getData(ciDataPath);
+
+  DataArray<float32>* ciDataArray = dynamic_cast<Float32Array*>(ciDataObject);
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:150]:
```suggestion
  auto& ciDataObject = dataGraph.getDataRefAs<Float32Array>(ciDataPath);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 3]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+  ConditionalSetValue filter;
+  Arguments args;
+  // Replace every value with a zero
+  args.insertOrAssign(ConditionalSetValue::k_ReplaceValue_Key, std::make_any<std::string>("0.0"));
+  args.insertOrAssign(ConditionalSetValue::k_ConditionalArrayPath_Key, std::make_any<DataPath>(DataPath({k_SmallIN100, k_EbsdScanData, "ConditionalArray"})));
+  args.insertOrAssign(ConditionalSetValue::k_SelectedArrayPath_Key, std::make_any<DataPath>(ciDataPath));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(dataGraph, args);
+  REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:168]:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 4]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(dataGraph, args);
+  REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:172]:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 5]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+
+  REQUIRE(RequireDataArrayEqualZero(*ciDataArray));
+}
+
+TEST_CASE("ConditionalSetValue: Test Algorithm Int8", "[ConditionalSetValue]")
+{
+  DataStructure dataGraph = UnitTest::CreateDataStructure();
+  DataPath ebsdScanPath = DataPath({k_SmallIN100, k_EbsdScanData});
+  DataPath geomPath = DataPath({k_SmallIN100, k_EbsdScanData, k_ImageGeometry});
+  std::shared_ptr<ImageGeom> imageGeometry = dataGraph.getSharedDataAs<ImageGeom>(geomPath);
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:182]:
```suggestion
  const ImageGeom& imageGeometry = dataGraph.getDataRefAs<ImageGeom>(geomPath);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 6]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+  complex::SizeVec3 imageGeomDims = imageGeometry->getDimensions();
+
+  DataPath ciDataPath = DataPath({k_SmallIN100, k_EbsdScanData, k_ConfidenceIndex});
+  DataObject* ciDataObject = dataGraph.getData(ciDataPath);
+
+  DataArray<float32>* ciDataArray = dynamic_cast<Float32Array*>(ciDataObject);
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:188]:
```suggestion
  auto& ciDataObject = dataGraph.getDataRefAs<Float32Array>(ciDataPath);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 7]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+  ConditionalSetValue filter;
+  Arguments args;
+  // Replace every value with a zero
+  args.insertOrAssign(ConditionalSetValue::k_ReplaceValue_Key, std::make_any<std::string>("0.0"));
+  args.insertOrAssign(ConditionalSetValue::k_ConditionalArrayPath_Key, std::make_any<DataPath>(DataPath({k_SmallIN100, k_EbsdScanData, "ConditionalArray"})));
+  args.insertOrAssign(ConditionalSetValue::k_SelectedArrayPath_Key, std::make_any<DataPath>(ciDataPath));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(dataGraph, args);
+  REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:206]:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp [hunk 8]
```diff
@@ -136,6 +136,82 @@ TEST_CASE("ConditionalSetValue: Test Algorithm", "[ConditionalSetValue]")
...
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(dataGraph, args);
+  REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ConditionalSetValueTest.cpp:210]:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]

**Pull request merged**